app/ui/src/app/core/providers/status-code-decoder-provider.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { StatusCodeDecoderService, MessageCode, LeveledMessage, StringMap } from '@syndesis/ui/platform';
+
+@Injectable()
+export class StatusCodeDecoderProviderService extends StatusCodeDecoderService {
+
+  // TODO eventually this would be in a separate json file, also better messages
+  private static messages = {
+    'SYNDESIS000': 'An unknown error has occurred', // generic message
+    'SYNDESIS001': 'There are parameter updates for this connection. Click <strong>Edit</strong> to update parameter settings.',
+    'SYNDESIS002': 'One or more properties have been updated or removed', // One or more properties have been added or removed
+    'SYNDESIS003': 'The connector associated with this connection has been deleted', // Connector has been deleted
+    'SYNDESIS004': 'The associated extension has been deleted', // Extension has been deleted
+    'SYNDESIS005': 'The associated action has been deleted', // Action has been deleted
+    'SYNDESIS006': 'One or more required properties is not set', // One or more required properties is not set
+    'SYNDESIS007': 'Secrets for this connection need to be updated', // Secrets update needed
+    'SYNDESIS008': 'There has a validation error', // Validation Error
+  } as StringMap<string>;

```

Comment:
I've got good news. Soon this will be moved to that JSON file you mentioned ;)

Reply:
Thought so!  :-)